pkg/apis/metal/types_infrastructure.go

```diff
@@ -15,9 +15,10 @@ type InfrastructureConfig struct {
 }
 
 type Firewall struct {
-	Size     string
-	Image    string
-	Networks []string
+	Size       string
+	Image      string
+	Networks   []string
+	RateLimits map[string]uint32
```

Comment:
You will also need to add this to the v1alpha1 infrastructure types, this is just the internal representation.